src/raftstore/store/snap.rs

```diff
@@ -61,223 +108,519 @@ impl Display for SnapKey {
     }
 }
 
-/// Name prefix for the self-generated snapshot file.
-const SNAP_GEN_PREFIX: &'static str = "gen";
-/// Name prefix for the received snapshot file.
-const SNAP_REV_PREFIX: &'static str = "rev";
+/// `SnapshotGenerator` is a trait that generates snapshot from `RocksDB` snapshot,
+/// the generated snapshot is used to replicate local db data to other remote raftstores.
+pub trait SnapshotGenerator {
+    fn generate(&mut self,
```

Comment:
It's not a Generator or Builder pattern. It just write a snapshot file to hard drive, and then server read the snapshot file through snapshot manager.